arkenv/src/test/kotlin/com/apurebase/arkenv/feature/cli/CliArgumentParserTest.kt

```diff
@@ -0,0 +1,37 @@
+package com.apurebase.arkenv.feature.cli
+
+import org.junit.jupiter.api.DynamicTest
+import org.junit.jupiter.api.TestFactory
+import strikt.api.expectThat
+import strikt.assertions.get
+import strikt.assertions.isEqualTo
+
+/**
+ * Verify the behavior of the [CliArgumentParser].
+ */
+internal class CliArgumentParserTest {
+
+    private data class TestCase(val name: String, val input: List<String>, val expected: String)
+
+    private val testCases = listOf(
+        TestCase("no quotes", listOf("hello", "world"), "hello"),
+        TestCase("double quotes", listOf("\"hello", "world\""), "hello world"),
+        TestCase("single quote inside value", listOf("D'vloper"), "D'vloper"),
+        TestCase("single quoted", listOf("'test", "expected'"), "test expected"),
+        TestCase("single quoted, containing singlq quote", listOf("'test", "ex'pected'"), "test ex'pected"),
```

Comment:
Maybe I'm thinking wrong. But shouldn't a case like this work also?
`TestCase("single quoted, containing singlq quote", listOf("'test", "ex\\'pected'"), "test ex'pected"),`

Reply:
It works without the backslash already, why would you need to provide the backslash?

Reply:
If you provide `--argument 'test\' expected'`, then it might make sense. 
I.e. if you want a quote at the end of a word (cli is split by space) but the value continues further.

Reply:
Oh, another case is: `--argument 'test'`
Should the output be `'test'` or `test`?

Reply:
I have no clue what the correct solution for this is. When using bash and cli tools I always just try stuff until it works :laughing: 

So maybe we can just leave it as is. Which means anything will work, as `main(args: List<String>)` is already splitt by spaces and you would just look at the first and last characters to merge them together.